inference/src/config.py

```
@@ -15,8 +15,18 @@ class Settings(BaseSettings):
+    # true only for trusted private networks (e.g. the bundled docker-compose setup).
+    allow_no_auth: bool = False
     model_config = {"env_prefix": "INFERENCE_"}
 
 
 settings = Settings()
+
+if not settings.auth_token and not settings.allow_no_auth:
+    raise RuntimeError(
+        "INFERENCE_AUTH_TOKEN is not set. Either set it to a strong secret or set "
+        "INFERENCE_ALLOW_NO_AUTH=true to explicitly opt into unauthenticated mode."
+    )
```

Ruff (0.15.12)

[warning] Avoid specifying long messages outside the exception class

(TRY003)